crates/polars-plan/src/logical_plan/aexpr/schema.rs

```diff
@@ -255,20 +255,6 @@ fn get_arithmetic_field(
             IDX_DTYPE
         },
         _ => {
-            match (left_ae, right_ae) {
```

Comment:
Instead of removing this whole functionality, can we branch this under numerical types? When dealing with numerical types, it is wasteful to upcast a whole column to a literal.